src/sentry/ratelimits/utils.py

```diff
@@ -50,17 +50,36 @@ def get_rate_limit_key(view_func: EndpointFunction, request: Request) -> str | N
     user_id = getattr(request_user, "id", None)
     is_sentry_app = getattr(request_user, "is_sentry_app", None)
 
-    request_access = getattr(request, "access", None)
-    org_id = getattr(request_access, "organization_id", None)
-
     ip_address = request.META.get("REMOTE_ADDR")
 
-    if is_sentry_app and org_id is not None:
+    request_auth = getattr(request, "auth", None)
```

Comment:
All of there `getattr()` calls are going to be difficult to maintain. It will be difficult about what is `None` when and messing with classes by name can make refactors difficult. I think you can do this with conditionals:
```py
if isinstance(request.auth, ApiToken):
    if request.user.is_sentry_app:
        category = "org"
        id = get_organization_id(request.auth.id)
    else:
        category = "user"
        id = request.auth.user_id

elif not isinstance(request.auth, ApiKey) and request.user:
    category = "user"
    id = request.user.id

elif ip_address is not None:
    category = "ip"
    id = ip_address
...
```

Reply:
And I'd move the query to managers.

`sentry/ratelimits/utils.py`:
```py
def get_organization_id(token_id: str) -> int | None:
    installation = SentryAppInstallation.objects.get_by_api_token(token_id).first()
    return installation.organization_id if installation else None
```
`sentry/models/sentryappinstallation.py`:
```py
class SentryAppInstallationForProviderManager(ParanoidManager):
    def get_by_api_token(self, token_id: str) -> QuerySet:
        return self.filter(sentryappinstallationtoken__api_token_id=token_id)
```